src/main/java/slimeknights/tconstruct/tables/client/inventory/module/TinkerStationButtonsScreen.java

```diff
@@ -0,0 +1,110 @@
+package slimeknights.tconstruct.tables.client.inventory.module;
+
+import com.mojang.blaze3d.matrix.MatrixStack;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.widget.Widget;
+import net.minecraft.client.gui.widget.button.Button;
+import net.minecraft.entity.player.PlayerInventory;
+import net.minecraft.inventory.container.Container;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.TranslationTextComponent;
+import slimeknights.tconstruct.library.client.Icons;
+import slimeknights.tconstruct.tables.client.SlotInformationLoader;
+import slimeknights.tconstruct.tables.client.inventory.ButtonItem;
+import slimeknights.tconstruct.tables.client.inventory.library.slots.SlotInformation;
+import slimeknights.tconstruct.tables.client.inventory.table.TinkerStationScreen;
+
+import java.util.List;
+
+public class TinkerStationButtonsScreen extends SideButtonsScreen {
+
+  protected final TinkerStationScreen parent;
+  protected int selected = 0;
+  private int style = 0;
+
+  public static final int WOOD_STYLE = 2;
+  public static final int METAL_STYLE = 1;
+
+  public TinkerStationButtonsScreen(TinkerStationScreen parent, Container container, PlayerInventory playerInventory, ITextComponent title) {
+    super(parent, container, playerInventory, title, TinkerStationScreen.COLUMN_COUNT, false);
+
+    this.parent = parent;
+  }
+
+  @SuppressWarnings("unchecked")
```

Comment:
Any reason this is unchecked? Only cases I see could also be written as `ButtonItem<?>`

Reply:
I can remove the suppress warning however, `ButtonItem<SlotInformation> buildScreenInfoButtonItem = (ButtonItem<SlotInformation>) button;` is unchecked as it's an unchecked cast for Button to `ButtonItem<SlotInformation>`

Reply:
Any way we can maintain a list of `ButtonItem<SlotInformation>` somewhere instead?